### PR TITLE
Refactoring duplicate code

### LINKED ISSUE #15: Remove warnings
In Javascript '===' is used to compare values as well as data type. In our project when we start the server there are various warnings that needed to be removed.
```
./src/components/form.jsx
[1]   Line 51:29:  Expected '===' and instead saw '=='  eqeqeq
[1]
[1] ./src/components/update.jsx
[1]   Line 51:29:  Expected '===' and instead saw '=='  eqeqeq
[1]
[1] ./src/store/reducer/user.js
[1]   Line 60:61:   Expected '===' and instead saw '=='  eqeqeq
[1]   Line 121:23:  Expected '===' and instead saw '=='  eqeqeq
[1]   Line 122:23:  Expected '===' and instead saw '=='  eqeqeq
[1]   Line 123:23:  Expected '===' and instead saw '=='  eqeqeq
[1]   Line 124:23:  Expected '===' and instead saw '=='  eqeqeq
[1]   Line 136:23:  Expected '===' and instead saw '=='  eqeqeq
[1]   Line 137:23:  Expected '===' and instead saw '=='  eqeqeq
[1]   Line 138:23:  Expected '===' and instead saw '=='  eqeqeq
[1]   Line 139:23:  Expected '===' and instead saw '=='  eqeqeq
[1]   Line 151:23:  Expected '===' and instead saw '=='  eqeqeq
[1]   Line 152:23:  Expected '===' and instead saw '=='  eqeqeq
[1]   Line 153:23:  Expected '===' and instead saw '=='  eqeqeq
[1]   Line 154:23:  Expected '===' and instead saw '=='  eqeqeq
[1]   Line 166:23:  Expected '===' and instead saw '=='  eqeqeq
[1]   Line 167:23:  Expected '===' and instead saw '=='  eqeqeq
[1]   Line 168:23:  Expected '===' and instead saw '=='  eqeqeq
[1]   Line 169:23:  Expected '===' and instead saw '=='  eqeqeq
```

### PR DESCRIPTION
**Pull request by mitalirs**
Fixes #15
Replaced '==' with '===' and removed related warnings

### CODE REVIEW

File: src/store/reducer/user.js [hunk 1]
```diff
@@ -118,10 +118,10 @@ function rootReducer(state = defaultState, action) {
...
       const newArray4 = state.subjdata.filter((user) => {
         return (
-          user.class1 == "9" ||
-          user.class2 == "9" ||
-          user.class3 == "9" ||
-          user.class4 == "9"
+          user.class1 === "9" ||
+          user.class2 === "9" ||
+          user.class3 === "9" ||
+          user.class4 === "9"
```
Comment (Namanl2001) [inline src/store/reducer/user.js:124]: Could you do something with these repetitive lines of code? For each `case C9 C10 C11 C12`

**Review by Namanl2001 — CHANGES_REQUESTED** [review]
Good Work!

File: src/store/reducer/user.js (hunk 1 above)
Comment (Namanl2001) [reply src/store/reducer/user.js:124]: Don't worry one of the mentors would help you with this soon.

Comment (Namanl2001) [reply src/store/reducer/user.js:124]: Hey! @sagar-barapatre please help her.

Comment (sagar-barapatre) [reply src/store/reducer/user.js:124]: Yeah! This works absolutely fine. I have checked it... 
Great job @mitalirs !!

Comment (Namanl2001) [reply src/store/reducer/user.js:124]: @mitalirs facing any issue? Discord project channels are open now.

Comment (mitalirs) [reply src/store/reducer/user.js:124]: Yes, I checked the channels. Actually, I wanted to add the shortened code to the pull request that I had created a few days back, but I am not sure about the best way of doing that. Once I figure that out I will update the PR.

**Review by sagar-barapatre — APPROVED** [review]

**Comment by mitalirs** [comment]
@sagar-barapatre, thank you for approving!
@Namanl2001, can you please review the PR?

**Review by Namanl2001 — CHANGES_REQUESTED** [review]
@mitalirs your changes are not working properly. Just tested your branch locally, adding screenshots to support my answer. Thanks.
image
image

**Comment by mitalirs** [comment]
@Namanl2001 I noticed that there were a few conflicts between my branch and the base branch. I have resolved the conflicts now. I am not very sure, but I think that might help. Could you please try re-running the code now?

**Comment by mitalirs** [comment]
Please let me know if the problem persists.

**Comment by Namanl2001** [comment]
you can also test them locally

**Comment by mitalirs** [comment]
I tried to set it up locally according to the instructions mentioned in the readme file, but I am getting errors while doing so

**Comment by Namanl2001** [comment]
Mentors would definitely help you out, head to the discord's project channel.

**Comment by sagar-barapatre** [comment]
@Namanl2001 The branch is running absolutely fine in my laptop as well as @mitalirs machine. But i don't know why is it that you are unable to see the changes. I even tested thing that once again today and its all well.

**Comment by Namanl2001** [comment]
Just checked it again. The subject filter is working fine but the same problem with the class filter. Also there are still some warnings in the console when the react server is starting, would be nice to remove them. 
Merge this branch with the latest master and if your changes work fine, then add the screenshots for reference.

**Comment by Namanl2001** [comment]
> add the screenshots for reference.

@mitalirs @sagar-barapatre What's the status of this PR?

**Comment by mitalirs** [comment]
> Just checked it again. The subject filter is working fine but the same problem with the class filter. Also there are still some warnings in the console when the react server is starting, would be nice to remove them.
> Merge this branch with the latest master and if your changes work fine, then add the screenshots for reference.

yes, I merged the branch some days ago. I am very sorry I couldn't reply before

**Comment by mitalirs** [comment]
> > add the screenshots for reference.
> 
> @mitalirs @sagar-barapatre What's the status of this PR?

Actually, I a still facing some setup related issues. I followed the steps as mentioned in the readme file, but when I try running it on localhost I see 'Initializing data' message displayed just before the footer. @sagar-barapatre is helping me resolve the problem

**Comment by sagar-barapatre** [comment]
> > add the screenshots for reference.
> 
> @mitalirs @sagar-barapatre What's the status of this PR?
Now it will not take much time as the setup issues are all solved. In a day or so changes will be reflected.

**Comment by mitalirs** [comment]
@Namanl2001, extremely sorry. There were some setup problems at my end because of the 'yarn' command. But now, I downloaded the zipped version of the code and tried to do it using npm as instructed by @sagar-barapatre and I could set up the project on my machine. You were right, the class filter was not working, sorry for that as well. That happened because the code did not convert the data type while comparing, i.e
it should have been: user.class1.toString() === action.type.slice(1)
and not: user.class1. === action.type.slice(1)
I have made the changes now
Here's the screenshot, with the class filter applied
picture2 (2)

**Review by sagar-barapatre — APPROVED** [review]
Yeah everything's fine. I tested this.

**Comment by Namanl2001** [comment]
@sagar-barapatre how have you tested this? Because it's still not proper, selecting any class shows this err:
image

**Comment by Namanl2001** [comment]
@mitalirs I've already commented my concerns above. thanks

**Comment by sagar-barapatre** [comment]
@Namanl2001 Nothing like that shows on my machine. I have added some SS to support this.
P.S. - I have setup the repo once again today morning and then showing you the results.

This shows when all classes are selected : 
Screenshot 2021-04-16 at 8 52 44 AM

This shows when class 9th is selected : 
Screenshot 2021-04-16 at 8 53 34 AM

This shows up when class 12th is selected : 
Screenshot 2021-04-16 at 8 54 28 AM

And finally this shows up when classes 10th and 11th are selected respectively : 
Screenshot 2021-04-16 at 8 55 33 AM


So please check that once. I don't think that it's the problem

**Comment by Namanl2001** [comment]
@sagar-barapatre you are not on this branch for sure because the screenshots you have attached has the new cards UI but as this branch is never rebased you should get the old cards UI. Also if you want to be sure you can verify the file changes on this PR with the files you are running.
AFAIK the best way to pull and test the pr's is to install the [github cli](https://github.com/cli/cli#installation) and run (like for this pr) `gh pr checkout 18` 

PS: @mitalirs I pulled and tested the changes and again got the same error as attached in the screenshot in my last review. Are you also getting the same error?

**Comment by mitalirs** [comment]
I have modified the code. Here are the screenshots:

Screenshot (345)
Screenshot (346)
Screenshot (347)
Screenshot (348)

@Namanl2001, @sagar-barapatre please review it once

**Review by Namanl2001 — CHANGES_REQUESTED** [review]
LGTM! Really great work @mitalirs All things are working as expected just one more change needed as mentioned.
PS: This pr is a combination of 2 issues `removing duplicate code` and `Removing warnings` so a `level3` label will be applied to it. Thanks

File: src/store/reducer/user.js [hunk 2]
```diff
@@ -56,8 +56,8 @@ function rootReducer(state = defaultState, action) {
         class: 'All',
       };
 
-    case 'PIN':
-      const newArray0 = state.all.filter(user => user.pin == action.pin);
+    case "PIN":
+      const newArray0 = state.all.filter((user) => user.pin === action.pin);
```
[the remark this answers is not in the document]
Comment (mitalirs) [reply src/store/reducer/user.js]: @Namanl2001, I have made the change as mentioned above, please review it

**Comment by Namanl2001** [comment]
There are two options to pass this check, both should work

- Try rebasing this branch
- Install prettier and eslint extensions in your VS code and use them in your changed files.

**Comment by mitalirs** [comment]
> There are two options to pass this check, both should work
> 
> * Try rebasing this branch
> * Install prettier and eslint extensions in your VS code and use them in your changed files.

I tried to rebase and then force pushed, but the problem persists...

**Comment by Namanl2001** [comment]
Open your terminal/ PowerShell (if on windows)
move to project root:
run 

- `yarn prettier --write '**/*.{js,jsx,css,html}' `
- `yarn eslint --fix '**/*.{js,jsx}'`

There might be some changes commit and push them.

**Comment by Namanl2001** [comment]
You can see the details of failing check there are still some formatting issues in `src/store/reducer/user.js` file. Try running the above commands once again.

**Comment by mitalirs** [comment]
> You can see the details of failing check there are still some formatting issues in `src/store/reducer/user.js` file. Try running the above commands once again.

yes, done

**Review by Namanl2001 — APPROVED** [review]
LGTM! @mitalirs Thanks for your patience and great work.

**Pull request merged**